filterdb/db.go

```diff
@@ -163,26 +163,38 @@ func putFilter(bucket walletdb.ReadWriteBucket, hash *chainhash.Hash,
 	return bucket.Put(hash[:], bytes)
 }
 
-// PutFilter stores a filter with the given hash and type to persistent
-// storage.
+// PutFilters stores a set of filters to persistent storage.
 //
 // NOTE: This method is a part of the FilterDatabase interface.
-func (f *FilterStore) PutFilter(filterData *FilterData) error {
-	return walletdb.Update(f.db, func(tx walletdb.ReadWriteTx) error {
+func (f *FilterStore) PutFilters(filterList ...*FilterData) error {
+	return walletdb.Batch(f.db, func(tx walletdb.ReadWriteTx) error {
```

Comment:
To make the best usage of `Batch`, we'd need to distinguish two types of errors,
- database error: the error which causes the whole `Batch` to fail, which means all other batched transactions will fail, such as a bucket is not found.
- transaction error: the error returned from a specific transaction, which only causes the current tx to fail.

[Here's a test](https://gist.github.com/yyforyongyu/fc82b46314d3ee7128dd8312182de785) I wrote before to check the `Batch` behavior. And we use [this pattern in lnd](https://github.com/lightningnetwork/lnd/blob/master/channeldb/payment_control.go#L130).